alcove/index/language.py

```
@@ -0,0 +1,299 @@
+    def detect(self, text: str) -> LanguageDetection:
+        sample = _sample_text(text)
+        if not sample:
+            return LanguageDetection(provider=self.provider)
+
+        try:
+            from langdetect import DetectorFactory, LangDetectException, detect_langs
+        except Exception as exc:
+            raise RuntimeError(
+                "ALCOVE_LANGUAGE_PROVIDER=langdetect requires the optional "
+                "`langdetect` package."
+            ) from exc
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)